src/core/components/ComponentHandler.ts

```diff
@@ -85,16 +85,10 @@ export abstract class ComponentHandler<
 			let value = this.loaderStrategy.load(filepath);
 			if (isThenable(value)) value = await value;
 
-			let didLoad = false;
 			for (const component of this.loaderStrategy.resolve(this, value)) {
 				const instance: TComponent = Reflect.construct(component, []);
 				promises.push(this.load(instance, filepath));
-
-				if (!didLoad) didLoad = true;
 			}
-
-			// Try unloading the file if no components were loaded.
-			if (!didLoad) this.loaderStrategy.unload?.(filepath);
```

Comment:
**Note:** This was removed as the logic is flaky as of this commit - components may have been resolved but not actually loaded as they were explicitly disabled. Furthermore, the benefit of this feature is dubious at best considering that is unlikely that a file in the directory given does not export any components.